server/controllers/addUser.js

```diff
@@ -0,0 +1,16 @@
+const { addUserQuery } = require('../database/queries');
+const { hashPassword, signupSchema } = require('../utils/validations');
+
+const addUser = (req, res, next) => {
+  const { error, value: { username, email, password } } = signupSchema.validate(req.body);
+
+  if (error) {
+    res.status(400).json({ msg: error.details[0].message });
+  } else {
+    hashPassword(password).then((hashed) => addUserQuery(username, email, hashed)
+      .then(() => next())
```

Comment:
The same as we did in the delete route, you should handle the response message here